nautobot_ssot_infoblox/diffsync/models/nautobot.py

```diff
@@ -63,29 +125,72 @@ def create(cls, diffsync, ids, attrs):
         )
         _ip.tags.add(create_tag_sync_from_infoblox())
+        if attrs.get("ext_attrs"):
+            cls.process_ext_attrs(diffsync=diffsync, ipaddr=_ip, extattrs=attrs["ext_attrs"])
         _ip.validated_save()
         return super().create(ids=ids, diffsync=diffsync, attrs=attrs)
 
     def update(self, attrs):
         """Update IPAddress object in Nautobot."""
-        _ipaddr = OrmIPAddress.objects.get(address=f"{self.address}/{self.prefix_length}")
+        _ipaddr = OrmIPAddress.objects.get(id=self.pk)
         if attrs.get("status"):
             _ipaddr.status = OrmStatus.objects.get(name=attrs["status"])
         if attrs.get("description"):
             _ipaddr.description = attrs["description"]
         if attrs.get("dns_name"):
             _ipaddr.dns_name = attrs["dns_name"]
+        if "ext_attrs" in attrs:
+            self.process_ext_attrs(diffsync=self.diffsync, ipaddr=_ipaddr, extattrs=attrs["ext_attrs"])
         _ipaddr.validated_save()
         return super().update(attrs)
 
     # def delete(self):
     #     """Delete IPAddress object in Nautobot."""
     #     self.diffsync.job.log_warning(self, message=f"IP Address {self.address} will be deleted.")
-    #     ip_ids = self.get_identifiers()
-    #     _ipaddr = OrmIPAddress.objects.get(address=f"{ip_ids['address']}/{ip_ids['prefix_length']}")
+    #     _ipaddr = OrmIPAddress.objects.get(id=self.pk)
     #     _ipaddr.delete()
     #     return super().delete()
 
+    @staticmethod
+    def process_ext_attrs(diffsync, ipaddr: OrmIPAddress, extattrs: dict):
+        """Process Extensibility Attributes into Custom Fields or link to found objects.
+
+        Args:
+            diffsync (object): DiffSync Job
+            ipaddr (OrmIPAddress): The IP Address that's being created or updated and needs processing.
+            extattrs (dict): The Extensibility Attributes to be analyzed and applied to passed `prefix`.
+        """
+        for attr, attr_value in extattrs.items():
+            if attr.lower() == "vrf":
+                try:
+                    ipaddr.vrf = OrmVRF.objects.get(name=attr_value)
+                except OrmVRF.DoesNotExist as err:
+                    diffsync.job.log_warning(
+                        f"Unable to find VRF {attr_value} for IP Address {ipaddr.address} found in Extensibility Attributes '{attr}'. {err}"
+                    )
+            if "role" in attr.lower():
+                if attr_value.lower() in IPAddressRoleChoices.as_dict():
+                    ipaddr.role = attr_value.lower()
+                else:
+                    diffsync.job.log_warning(
+                        f"Unable to find Role {attr_value} for IP Address {ipaddr.address} found in Extensibility Attributes '{attr}'."
+                    )
+            if attr.lower() in ["tenant", "dept", "department"]:
+                try:
+                    ipaddr.tenant = OrmTenant.objects.get(name=attr_value)
```

Comment:
Could we clean this up a bit and standardize into a helper function? And handle exceptions as they occur since there is a bit of duplicate code among the models.

I was thinking it would accept the query params ({"name": attr_value}), the ORM object it needs to perform the `.get()` on and the error handling.

Reply:
I've refactored the 'process_ext_attrs' function to be a helper like requested and confirmed it's working.